lib/Sema/DerivedConformances.cpp

```diff
@@ -77,6 +77,10 @@ bool DerivedConformance::derivesProtocolConformance(DeclContext *DC,
         // conformance.
       case KnownProtocolKind::Equatable:
         return canDeriveEquatable(DC, Nominal);
+      
+      case KnownProtocolKind::Comparable:
+        return !enumDecl->hasPotentiallyUnavailableCaseValue()
+            && canDeriveComparable(DC, Nominal); // why do y’all pass wide `NominalTypeDecl*` value when u could pass downcast `EnumDecl*` value?
```

Comment:
You can refine the type of that parameter.

Reply:
done